lib/rubocop/cop/rspec/indexed_let.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Do not set up test data using indexes (e.g., `item_1`, `item_2`).
+      #
+      # It makers reading the test harder because it's not clear what exactly
+      # is tested by this particular example.
+      #
+      # @example
+      #   # bad
+      #   let(:item_1) { create(:item) }
+      #   let(:item_2) { create(:item) }
+      #
+      #   let(:item_1) { create(:item, visible: true) }
+      #   let(:item_2) { create(:item, visible: false) }
+      #
+      #   before do
+      #     item_1 = create(:item)
+      #     item_2 = create(:item)
+      #   end
+      #
+      #   # good
+      #   let(:items) { create_list(:item, 2) }
+      #
+      #   let(:visible_item) { create(:item, visible: true) }
+      #   let(:invisible_item) { create(:item, visible: false) }
+      #
+      #   before do
+      #     create_list(:item, 2)
+      #   end
+      #
+      class IndexedLet < Base
+        MSG = 'This block declares indexed `let` statements: ' \
+              '%<duplicates>s. Please give them meaningful names,' \
+              ' use create_list or move creation to the `before` block.'
+
+        # @!method describe_or_context(node)
+        def_node_matcher :describe_or_context, <<~PATTERN
+          (block (send nil? {:describe | :context} ...) _ (begin $...))
+        PATTERN
+
+        # @!method any_let(node)
+        def_node_matcher :any_let, <<~PATTERN
+          (block (send nil? {:let | :let! | :let_it_be} (sym $_)) _ ...)
+        PATTERN
+
```

Comment:
```suggestion
```
We have `let?` node matcher.

Reply:
That would also match blockpass. It probably makes sense to add a spec for blockpass just in case.

Reply:
Fixed